lib/Parser.js

```diff
@@ -297,6 +297,29 @@ Parser.prototype.initializeEvaluating = function() {
 			}
 			return new BasicEvaluatedExpression().setString(result).setRange(expr.range);
 		});
+		this.plugin("evaluate TemplateLiteral", function(node) {
+			var i, quasis = node.quasis, exprs = [];
+			for (i = 0; i < quasis.length; i++) {
+				exprs.push(this.evaluateExpression(quasis[i]).cooked);
+				// possible improvement: evaluate node.expressions[i-1]
+				// and append to previous quasis if has a set value
+			}
+			return new BasicEvaluatedExpression().setTemplateString(exprs).setRange(node.range);
+		});
+		this.plugin("evaluate TaggedTemplateExpression", function(node) {
+			if (this.evaluateExpression(node.tag).identifier !== 'String.raw') return;
+			var i, quasis = node.quasi.quasis, exprs = [];
+			for (i = 0; i < quasis.length; i++) {
+				exprs.push(this.evaluateExpression(quasis[i]).raw);
+			}
+			return new BasicEvaluatedExpression().setTemplateString(exprs).setRange(node.range);
+		});
+		this.plugin("evaluate TemplateElement", function(node) {
+			return {
+				raw: new BasicEvaluatedExpression().setString(node.value.raw).setRange(node.range),
+				cooked: new BasicEvaluatedExpression().setString(node.value.cooked).setRange(node.range)
+			};
```

Comment:
We don't really need this hook. It's doesn't fit into the `evaluate *` hook style. Just more the code into the two above hooks.